### DISCUSSION TITLE
how do i compile the program

### DISCUSSION THREAD

**Discussion by dongzheng9** [LizardByte/community#153] [Sunshine] [answered]
I want to try ipv6 so I downloaded  cgutman  source code But I'm sorry I won't compile
I try to use loki's compilation method [loki-47-6F-64](https://github.com/loki-47-6F-64/sunshine) 
Compilation:
git clone https://github.com/cgutman/LB_Sunshine.git --recursive
cd sunshine && mkdir build && cd build
cmake -G"Unix Makefiles" ..
mingw32-make

But there was a mistake It cannot take effect here Would you like to know if there is a suitable way to compile into exe or zip files
esteem it a favor  thx

**Comment by ReenigneArcher** [comment, marked as answer]
Instructions are in our docs.

https://docs.lizardbyte.dev/projects/sunshine/en/latest/building/build.html

**Reply by demothus** [reply]
I'm not able to get Sunshine to build using the document, I've tried following it at least 10 times on a virtual machine (Windows 10 x64 bit) but continue to get the following errors during the build (please see attached).

![Build Error](https://github.com/LizardByte/Sunshine/assets/9018728/b2a3e44f-650b-4937-8f75-2267e6d75ce9)

**Reply by ReenigneArcher** [reply]
Use nightly branch.

**Reply by demothus** [reply]
I've tried both nightly and master multiple times.

I have zero programming experience, so maybe I'm not selecting the branch correctly? I've been using "git switch nightly" and then verifying everything is up-to-date using "git pull", but I'm still getting the same errors. :(

**Reply by ReenigneArcher** [reply]
Did you update submodules after switching branches?

**Reply by demothus** [reply]
Sorry for my ignorance, how do I do that?

**Reply by ReenigneArcher** [reply]
```bash
cd third-party
git submodule update --init --recursive
```

**Reply by demothus** [reply]
That did it!

That's not anywhere in the documentation, which would explain why I was having problems...is there a way to make sure the project is fully up-to-date instead of running that submodule update command on every single directory, or is it only needed for the third-party directory?

I really appreciate your help! :)

**Reply by ReenigneArcher** [reply]
It's recursive so you don't need to do it on every directory, it does that automatically.
